bibliopixel/project/types/colors.py

```diff
@@ -0,0 +1,22 @@
+import functools
+from . import color
+
+USAGE = """Colors is a list of colors.  Each color can be:
+
+* A list of three numbers: [0, 0, 0] or [255, 0, 255].
+* A single number which represents a brightness/gray level: 0, 255, 127
+* A string:  "red", "yellow", "gold" naming a color from ...colors.COLORS.
+
+All numbers must be in the range [0, 256) - 0 <= x < 256
+"""
+
+
+@functools.singledispatch
+def make(c):
+    raise ValueError("Don't understand type %s" % type(c), USAGE)
+
+
+@make.register(tuple)
+@make.register(list)
+def _(c):
+    return [color.make(i) for i in c]
```

Comment:
It's `singledispatch`!

https://docs.python.org/3/library/functools.html#functools.singledispatch

A really nice facility, but rarely the correct solution to a problem.  However, this one problem it's perfect for (and is 90%+ of the time I've seen it used...)